tidy3d/components/medium.py

```diff
@@ -2045,6 +2045,92 @@ def to_medium(self) -> Medium:
             frequency_range=self.frequency_range,
         )
 
+    @classmethod
+    def lo_to_eps_model(
+        cls,
+        poles: Tuple[Tuple[float, float, float, float], ...],
+        eps_inf: pd.PositiveFloat,
+        frequency: float,
+    ) -> complex:
+        """Evaluate eps model for a given set of LO-TO coefficients."""
+        omega = 2 * np.pi * frequency
+        eps = eps_inf
+        for (omega_lo, gamma_lo, omega_to, gamma_to) in poles:
+            eps *= omega_lo**2 - omega**2 - 1j * omega * gamma_lo
+            eps /= omega_to**2 - omega**2 - 1j * omega * gamma_to
+        return eps
+
+    @classmethod
+    def from_lo_to(
+        cls, poles: Tuple[Tuple[float, float, float, float], ...], eps_inf: pd.PositiveFloat = 1
+    ) -> PoleResidue:
+        """Construct a pole residue model from the LO-TO form
+        (longitudinal and transverse optical modes).
+        The LO-TO form is :math:`\\epsilon_\\infty \\prod_{i=1}^l \\frac{\\omega_{LO, i}^2 - \\omega^2 - i \\omega \\gamma_{LO, i}}{\\omega_{TO, i}^2 - \\omega^2 - i \\omega \\gamma_{TO, i}}` as given in the paper:
+
+            M. Schubert, T. E. Tiwald, and C. M. Herzinger,
+            "Infrared dielectric anisotropy and phonon modes of sapphire,"
+            Phys. Rev. B 61, 8187 (2000).
+
+        Parameters
+        ----------
+        poles : Tuple[Tuple[float, float, float, float], ...]
+            The LO-TO poles, given as list of tuples of the form
+            (omega_LO, gamma_LO, omega_TO, gamma_TO).
+
+        Returns
+        -------
+        :class:`.PoleResidue`
+            The pole residue equivalent of the LO-TO form provided.
+        """
+
+        omegas_lo, gammas_lo, omegas_to, gammas_to = (np.array(param) for param in zip(*poles))
+
+        # discriminants of quadratic factors of denominator
+        discs = 2 * np.emath.sqrt((gammas_to / 2) ** 2 - omegas_to**2)
+
+        # require nondegenerate TO poles
+        if len({(omega_to, gamma_to) for (_, _, omega_to, gamma_to) in poles}) != len(poles) or any(
+            disc == 0 for disc in discs
```

Comment:
When will the 1st condition occur?

Reply:
The first condition will occur when a TO pole appears repeatedly within the provided list of poles. Then, the set will be smaller than the list of poles provided.